filebeat/fileset/factory.go

```diff
@@ -0,0 +1,86 @@
+package fileset
+
+import (
+	"github.com/elastic/beats/libbeat/cfgfile"
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/logp"
+	"github.com/mitchellh/hashstructure"
+
+	"github.com/elastic/beats/filebeat/channel"
+	"github.com/elastic/beats/filebeat/prospector"
+	"github.com/elastic/beats/filebeat/registrar"
+)
+
+// Factory is a factory for registrars
```

Comment:
Is this really a factory of registrars? Or did you mean a factory of modules or prospectors?